lib/ansible/modules/net_tools/ldap/ldap_passwd.py

```diff
@@ -117,7 +117,7 @@ def passwd_set(self):
 
         # Change the password (or throw an exception)
         try:
-            self.connection.passwd_set(self.dn, None, self.passwd)
```

Comment:
The question is, has this interface ever existed? 

if yes, then I would suggest:
~~~diff
-             self.connection.passwd_set(self.dn, None, self.passwd)
+             try:
+                 self.connection.passwd_set(self.dn, None, self.passwd)
+             except AttributeError:
+                 self.connection.passwd(self.dn, None, self.passwd)
~~~

Reply:
Here's the [documentation](https://www.python-ldap.org/en/latest/reference/ldap.html?highlight=passwd#ldap.LDAPObject.passwd) I referred to when making this change

Reply:
@drewburr Indeed not!  This was `passwd_s` in [the version we use in production](https://github.com/hashbang/admin-tools/blob/master/library/ldap_passwd.py#L181), and it looks like the bug creeped in with the modifications I was asked to do to ship it in Ansible.

Reply:
There is a slight issue with your proposed fix (`.passwd()` isn't a synchronous interface), I will fix that in a jiffy.